compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -107,13 +110,17 @@ pub struct UnresolvedTrait {
 pub struct UnresolvedTraitImpl {
     pub file_id: FileId,
     pub module_id: LocalModuleId,
-    pub trait_id: Option<TraitId>,
     pub trait_generics: Vec<UnresolvedType>,
     pub trait_path: Path,
     pub object_type: UnresolvedType,
     pub methods: UnresolvedFunctions,
     pub generics: UnresolvedGenerics,
     pub where_clause: Vec<UnresolvedTraitConstraint>,
+
+    // These fields are filled in later
```

Comment:
```suggestion
    // These fields are filled in later during elaboration
```
Nit, feel free to add more as you like. Saying "later" makes sense to me when declaring an object, but in the type definition it makes me think "later when?"